### PR TITLE
Custom notifications

### LINKED ISSUE #142: Allow Manual Notifications/Emails
As stated by @anondev-sudo -
"Admins [could send] notifications to users on dashboard or using API. There should be also that you can choose if you want to send just notification on dashboard or just email notification or both."

_Originally posted by @anondev-sudo in https://github.com/ControlPanel-gg/dashboard/issues/136#issuecomment-882081973_

### PR DESCRIPTION
**Pull request by j122j**
This pull request adds custom notifications.
Both website and email notifications can be sent using the admin dashboard or using API.

Checklist:
- [x] Add admin UI.
- [x] Add API endpoints.

API Endpoints:
- GET /api/notifications/{user}
- GET /api/notifications/{user}/{notification}
- POST /api/notifications
- DELETE /api/notifications/{user}
- DELETE /api/notifications/{user}/{notification}

Closes #142 #106

### CODE REVIEW

**Comment by j122j** [comment]
Admin UI Preview
5014417d

File: resources/views/admin/users/notifications.blade.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                    <div class="card">
+                        <div class="card-body">
+                            <form action="{{route('admin.users.notifications')}}" method="POST">
+                                @csrf
+                                @method('POST')
+
+                                <div class="form-group">
+                                    <label>Users</label><br>
+                                    <input id="all" name="all"
+                                           type="checkbox"
```
Comment (AVMG20) [inline resources/views/admin/users/notifications.blade.php:38]: add value="1" so the bool validation works

File: resources/views/admin/users/notifications.blade.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    <!-- MAIN CONTENT -->
+    <section class="content">
+        <div class="container-fluid">
+
+            <div class="row">
+                <div class="col-lg-12">
+                    <div class="card">
+                        <div class="card-body">
+                            <form action="{{route('admin.users.notifications')}}" method="POST">
```
Comment (AVMG20) [inline resources/views/admin/users/notifications.blade.php:31]: form tag is not closed

**Review by AVMG20 — CHANGES_REQUESTED** [review]
Excellent PR, 2 minor issues but overall very good code base and functionality.

**Pull request merged**